apps/explorer/src/components/transaction-card/RecentTxCard.tsx

```diff
@@ -2,180 +2,83 @@
 }: Props) {
     const [paused, setPaused] = useState(false);
-    const [txPerPage, setTxPerPage] = useState(
-        initialTxPerPage || NUMBER_OF_TX_PER_PAGE
-    );
+    const [limit, setLimit] = useState(initialLimit);
 
     const rpc = useRpcClient();
-    const [searchParams, setSearchParams] = useSearchParamsMerged();
 
-    const [pageIndex, setPageIndex] = useState(
-        parseInt(searchParams.get('p') || '1', 10) || 1
+    const countQuery = useQuery(['transactions', 'count'], () =>
+        rpc.getTotalTransactionNumber()
     );
 
-    const handlePageChange = useCallback(
-        (newPage: number) => {
-            setPageIndex(newPage);
-            setSearchParams({
-                p: newPage.toString(),
-            });
-        },
-        [setSearchParams]
-    );
-
-    const countQuery = useQuery(
-        ['transactions', 'count'],
-        () => rpc.getTotalTransactionNumber(),
-        {
-            refetchInterval: paused ? false : TRANSACTION_POLL_TIME,
-        }
-    );
+    const pagination = usePaginationStack();
+    const refetching = !pagination.cursor && !paused;
 
     const transactionQuery = useQuery(
-        ['transactions', { total: countQuery.data, txPerPage, pageIndex }],
-        async () => {
-            const { data: count } = countQuery;
-
-            if (!count) {
-                throw new Error('No transactions found');
-            }
-
-            // If pageIndex is greater than maxTxPage, set to maxTxPage
-            const maxTxPage = Math.ceil(count / txPerPage);
-            const pg = pageIndex > maxTxPage ? maxTxPage : pageIndex;
-
-            return getRecentTransactions(rpc, count, txPerPage, pg);
-        },
+        ['transactions', { limit, cursor: pagination.cursor }],
+        async () =>
+            rpc.queryTransactions({
+                order: 'descending',
+                cursor: pagination.cursor,
+                limit,
+                options: {
+                    showEffects: true,
+                    showBalanceChanges: true,
+                    showInput: true,
+                },
+            }),
         {
             enabled: countQuery.isFetched,
             keepPreviousData: true,
+            refetchInterval: refetching ? TRANSACTION_POLL_TIME : false,
         }
     );
 
     const recentTx = useMemo(
         () =>
             transactionQuery.data
-                ? genTableDataFromTxData(transactionQuery.data)
+                ? genTableDataFromTxData(transactionQuery.data.data)
                 : null,
         [transactionQuery.data]
     );
 
-    const stats = {
-        count: countQuery?.data || 0,
-        stats_text: 'Total Transactions',
-        loadState: statusToLoadState[countQuery.status],
-    };
-
     const handlePauseChange = () => {
-        // If we were paused, immedietly refetch:
         if (paused) {
-            countQuery.refetch();
+            // If we were paused, and on the first page, immedietly refetch:
```

Comment:
I know you didn't write this but grammar 👮🏼 

```suggestion
            // If we were paused, and on the first page, immediately refetch:
```

Reply:
I think I did write this beforehand thought 🤦  I can never spell this work right.